crates/gitbutler-branch-actions/src/actions.rs

```diff
@@ -133,6 +133,7 @@ pub fn delete_local_branch(project: &Project, refname: &Refname, given_name: Str
 #[instrument(skip(project), err(Debug))]
 pub fn get_base_branch_data(project: &Project) -> Result<BaseBranch> {
     let ctx = CommandContext::open(project)?;
+    update_base_branch(project)?;
     base::get_base_branch_data(&ctx)
 }
```

Comment:
Is this safe?

Doesn't this mean that we could be accidentally rebasing peoples branches when they don't expect it?

Reply:
Oh nice catch! I'm not sure 🤔.
I'll check

Reply:
Yeah, this means any time you refresh the app, or perform an action that re-fetches the sate of the base branch we're going to perform that update action.

We absolutely can't merge this in.

Reply:
Gotcha 👍 

I guess then the fix should go into displaying the the update workspace update button for the situation I outlined in the PR description

Reply:
The issue here, is that we have no way of representing `master` (or your target commit) is ahead of `origin/master` in GitButler. We currently don't have mechanisms for merging branches into your local `master` either, so the only way to get into this state is by doing some fiddling via the git CLI.

We probably need to discuss how we want to handle having local `master` ahead of `origin/master` CC. @krlvi 

One possible solution would be to have a button which when clicked fixes any divergence between local `master` and `origin/master`. We ought to have extra warning so the user knows what is happening, because "Update" is for sure not what we'd be doing in that scenario.

Another, would be to add in a "push" button to push up that extra commit on local `master`

(or perhaps both)

Reply:
I see, yeah that makes sense.

Reply:
@estib-vega A great rule of physics of *this* universe is also that functions that indicate they are read-only, should never, ever also mutate state on disk. Right now, there is a couple of even massively important functions (I am looking at you, `list_virtual_branches()`) which violate physics essentially, and it's definitely something to roll-back and fix as well. Until then, it will let the cat out of the box occasionally :D.

In Rust, violating this rule isn't easily possible (even though it can be side-stepped), but since all of the code here is based on state that is read from disk, Rust isn't here to safe the day, unfortunately. In `list_virtual_branches()` you will also find the `perm:` parameter which acts as a marker to indicate if a method is allowed to read or write. Behind that is an actual file-based lock so no function marked that way can execute if another one is already running. However, due to this lock being quite costly, it's only acquired for heavier functions, not for those which essentially just read a file on disk (or write it).

I hope that helps.

Reply:
It does! Thanks for the pointers :)